packages/amplify-app/src/xcodeHelpers.js

```diff
@@ -15,42 +40,143 @@ async function getXcodeProjectDir() {
+ */
+function getOrCreateGroup(project, name) {
+  let group = getGroupByName(project, name);
+  if (!group) {
+    group = project.addPbxGroup([], name, '.');
+    const rootGroup = getRootGroup(project);
+    rootGroup.pbxGroup.children = [{ value: group.uuid, comment: group.pbxGroup.name }, ...rootGroup.pbxGroup.children];
+  }
+  return group;
+}
+
+/**
+ * @private
+ * @param {PBXGroupRef} group
+ * @param {string} name
+ * @returns {boolean}
+ */
+function groupHasFile(group, name) {
+  return group.pbxGroup.children.filter(child => child.comment === name).length > 0;
+}
+
+/**
+ * @public
+ * @return {Promise<void>}
+ */
+async function addDataStoreFiles() {
+  const projectDir = getXcodeProjectDir();
+  const rootDir = path.resolve(projectDir, '..', '..');
+  const project = xcode.project(projectDir);
+  return new Promise((resolve, reject) => {
+    const schemaFilePattern = path.join(rootDir, 'amplify', 'backend', 'api', '*', 'schema.graphql');
+    const [schemaFile] = glob.sync(schemaFilePattern);
+    if (!schemaFile) {
+      reject(new Error('schema.graphql file not found'));
+      return;
+    }
+
+    project.parse(error => {
+      if (error) {
+        reject(error);
+        return;
       }
 
-      fs.writeFileSync(projectPath, myProj.writeSync());
-      if (err) {
-        reject(err);
+      const rootGroup = getRootGroup(project);
+      if (rootGroup == null || typeof rootGroup.uuid !== 'string') {
+        reject(new Error('Could not find root group of Xcode project'));
+        return;
       }
-      resolve();
-    })
-  );
+
+      try {
+        // step 1: add generated models
+        const modelsFilePattern = path.join(rootDir, 'amplify', 'generated', 'models', '*.swift');
+        const modelFiles = glob.sync(modelsFilePattern).map(file => {
+          return path.relative(rootDir, file);
+        });
+        if (modelFiles && modelFiles.length > 0) {
+          const modelsGroup = getOrCreateGroup(project, 'AmplifyModels');
+          modelFiles.forEach(file => {
+            const { base: filename } = path.parse(file);
+            if (!groupHasFile(modelsGroup, filename)) {
+              console.log(`adding model source file... ${file}`);
+              project.addSourceFile(file, {}, modelsGroup.uuid);
+            }
+          });
+        }
+
+        // step 2: add configuration, schema and other amplify resources
+        const amplifyConfigGroup = getOrCreateGroup(project, 'AmplifyConfig');
+
+        // add the amplifytools config file
+        project.addFile('./amplifytools.xcconfig', amplifyConfigGroup.uuid);
+
+        // adding resources (i.e. files that are added to the app bundle)
+        project.addPbxGroup([], 'Resources', 'Resources', '"<group>"');
+        project.addResourceFile('amplifyconfiguration.json', null, amplifyConfigGroup.uuid);
+        project.addResourceFile('awsconfiguration.json', null, amplifyConfigGroup.uuid);
+        project.removePbxGroup('Resources');
```

Comment:
Why are we removing this?

Reply:
This is the same logic we had before, just moved from the `index.js` here.

Reply:
The Resources group is added so that we can add the resource files, the group is unneeded outside of this operation so its removed. It's a limitation of the cordova xcode library that the group is required in the first place

Reply:
`amplify/backend/api/*/{schema/*.graphql,schema.graphql}` should do it